src/main/java/com/aws/greengrass/lifecyclemanager/Kernel.java

```diff
@@ -354,117 +356,140 @@ public Topics findServiceTopic(String serviceName) {
      * @return found service or null
      * @throws ServiceLoadException if service cannot load
      */
-    @SuppressWarnings(
-            {"UseSpecificCatch", "PMD.AvoidCatchingThrowable", "PMD.AvoidDeeplyNestedIfStmts", "PMD.ConfusingTernary"})
     public GreengrassService locate(String name) throws ServiceLoadException {
-        return context.getValue(GreengrassService.class, name).computeObjectIfEmpty(v -> {
-            Topics serviceRootTopics = findServiceTopic(name);
-
-            Class<?> clazz = null;
-            if (serviceRootTopics != null) {
+        return context.getValue(GreengrassService.class, name).computeObjectIfEmpty(v ->
+                locateGreengrassService(v, name, this::locate));
+    }
 
-                // Try locating all the dependencies first so that they'll all exist prior to their dependant.
-                // This is to fix an ordering problem with plugins such as lambda manager. The plugin needs to be
-                // located *before* the dependant is located so that the plugin has its jar loaded into the classloader.
-                Topic dependenciesTopic = serviceRootTopics.findLeafChild(SERVICE_DEPENDENCIES_NAMESPACE_TOPIC);
-                if (dependenciesTopic != null && dependenciesTopic.getOnce() instanceof Collection) {
-                    try {
-                        for (Pair<String, DependencyType> p : GreengrassService
-                                .parseDependencies((Collection<String>) dependenciesTopic.getOnce())) {
-                            locate(p.getLeft());
-                        }
-                    } catch (ServiceLoadException | InputValidationException e) {
-                        throw new ServiceLoadException("Unable to load service " + name, e);
-                    }
-                }
+    /**
+     * Locate a GreengrassService by name in the kernel context, or return BrokenService if service cannot load.
+     *
+     * @param name name of the service to find
+     * @return found service
+     */
+    public GreengrassService locateIgnoreError(String name) {
+        return context.getValue(GreengrassService.class, name).computeObjectIfEmpty(v -> {
+            try {
+                return locateGreengrassService(v, name, this::locateIgnoreError);
+            } catch (ServiceLoadException e) {
+                logger.atError().log("Cannot load service", e);
+                return new BrokenService(config.lookupTopics(DEFAULT_VALUE_TIMESTAMP, SERVICES_NAMESPACE_TOPIC, name),
+                        e);
```

Comment:
Do we only want to do this when the service actually exists, does this method check for that?

What I mean is that if we lookup a service which doesn't exist, then I still want the exception and not a "BrokenService".

Reply:
By `service which doesn't exist`, do you mean the config node is missing in config files? In this case, BrokenService is also created. Otherwise it's still a failure case which can brick device if nucleus restarts.

Reply:
Right, when the node doesn't exist, that doesn't mean that it is broken, it means that it doesn't exist. Those are different cases.

Reply:
Is it better to rename BrokenService to UnloadableService?